CodeConverter/CSharp/QueryConverter.cs

```diff
@@ -144,19 +155,13 @@ private async Task<CSharpSyntaxNode> ConvertQuerySegmentsAsync(IEnumerable<(Queu
             return invocationExpressionSyntax;
         }
 
-        private async Task<CSharpSyntaxNode> ConvertQueryWithContinuationsAsync(Queue<(SyntaxList<CSSyntax.QueryClauseSyntax>, VBSyntax.QueryClauseSyntax)> queryContinuation, CSSyntax.FromClauseSyntax fromClauseSyntax)
-        {
-            var subQuery = await ConvertQueryWithContinuationAsync(queryContinuation, fromClauseSyntax.Identifier.WithoutSourceMapping());
-            return subQuery != null ? SyntaxFactory.QueryExpression(fromClauseSyntax, subQuery) : fromClauseSyntax.Expression;
-        }
-
-        private async Task<CSSyntax.QueryBodySyntax> ConvertQueryWithContinuationAsync(Queue<(SyntaxList<CSSyntax.QueryClauseSyntax>, VBSyntax.QueryClauseSyntax)> querySectionsReversed, SyntaxToken reusableCsFromId)
+        private async Task<CSSyntax.QueryBodySyntax> ConvertQueryWithContinuationAsync(Queue<(SyntaxList<CSSyntax.QueryClauseSyntax>, VBSyntax.QueryClauseSyntax)> querySectionsReversed, SyntaxToken? reusableCsFromId)
         {
             if (!querySectionsReversed.Any()) return null;
             var (convertedClauses, clauseEnd) = querySectionsReversed.Dequeue();
-
+            reusableCsFromId ??= convertedClauses.OfType<CSSyntax.FromClauseSyntax>().FirstOrDefault()?.Identifier.WithoutSourceMapping();
```

Comment:
Ideally we'd take a step back and understand how VB determines which id to select. It's possible that an up front analysis would be worthwhile to tidy this mess up.